### DISCUSSION TITLE
Support plum-like inheritance dispatch.

### DISCUSSION THREAD

**Discussion by coady** [coady/multimethod#106] [Polls]
[Plum's comparison](https://beartype.github.io/plum/comparison.html#careful-synergy-with-oop) with `multimethod` cites this example:
```python
from multimethod import multimethod


class A:
    def f(self, x):
        return "fallback"
        

class B(A):
    @multimethod
    def f(self, x: int):
        return x
```
```python
>>> b = B()

>>> b.f(1)
1

>>> b.f("1")
DispatchError: ('f: 0 methods found', (<class '__main__.B'>, <class 'str'>), [])
```

A normal method dispatches based on its presence, not on its arguments. So to me this is ambiguous, and a reason to use `multimethod` functions instead of methods. The ambiguity comes from whether one pictures the dispatch occurring "twice", once on `self` and then again on all the args.

But it could be trivially supported, and backwards-compatible (assuming no one was relying on the dispatch error). So opening it up for discussion.

**Comment by coady** [comment]
Should also add there is a current solution: explicitly register the methods.
```python
class A:
    def f(self, x):
        return "fallback"
        

class B(A):
    f = multimethod(A.f)

    @multimethod
    def f(self, x: int):
        return x
```

**Reply by snwnde** [reply]
The current solution appears good to me. This even works when the user has no control on the parent class `A` (see https://github.com/coady/multimethod/discussions/75#discussioncomment-4608934).

What is more challenging is when dealing with classmethods. How could one apply last `classmethod` (or `staticmethod`) when `f` is already a classmethod in the parent class?

**Reply by coady** [reply]
```python
class A:
    @classmethod
    def f(cls, x):
        return "fallback"
        

class B(A):
    f = multimethod(A.f.__func__)

    @classmethod
    @multimethod
    def f(cls, x: int):
        return x
```
